### ISSUE TITLE
Reflection on SPRINT 2

### ISSUE THREAD

**Issue by Umkhaled** [Umkhaled/genetic-variant-annotator#1]
In this issue, I am reflecting on the progress made during SPRINT 2, how I met the Minimum Viable Product (MVP), and the challenges I faced.

 How I Met the MVP
 Focused on implementing key features like setting up the environment, building the Flask API, and automating tests.
 Managed to complete the MVP by ensuring the API could process requests and return mock data.
Verified functionality through testing and debugging.

Challenges and Learnings
 Encountered challenges like setting up the correct environment and managing dependencies.
 Overcame them by adjusting the `environment.yml` and ensuring all dependencies were properly documented.
Learned more about using `pytest` for testing and handling errors in a Flask application.

Evidence of Task Management
 Managed tasks and tracked progress through Git commits.
Documented the process and changes made to the project through this issue for transparency.
 Worked independently but kept detailed notes on progress.

**Comment by Umkhaled** [comment]
During SPRINT 2, I worked independently on tasks such as setting up the environment using `conda`, building the Flask API, and writing automated tests. Even though I didn’t have a team to collaborate with, I simulated the process of a team environment by:

Creating and tracking tasks in Git.
Adjusting development processes as needed to meet the MVP.
 Focusing on implementing feedback from my supervisor regarding the structure of the project.

 Adjustments Made
 Improved the environment setup by refining the `environment.yml`.
Moved the `app.py` file into a dedicated folder for better organization.
Added automated tests using `pytest` and resolved errors during the testing phase.

final Thoughts
 I managed to meet the MVP requirements by focusing on core functionalities like API responses and environment setup.
 I learned a lot about managing a project solo and using Git for documentation and task tracking.
